app/api/crud.py

```diff
@@ -15,8 +15,7 @@ def build_combined_response(
     content = {"errors": node_errors, "responses": cross_node_results}
 
     if node_errors:
-        # TODO: Use logger instead of print. For example of how to set this up for FastAPI, see https://github.com/tiangolo/fastapi/discussions/8517
-        print(
+        logging.error(
```

Comment:
How about using the warning level for this summary statement instead of error, 
- since the detailed failure message from each node is also logged as a warning
- so that a user is not alarmed by the error (otherwise, it appears like it is an error with the f-API itself even though it is not)

Reply:
Not sure I agree. This will just go to the log, the app will continue working. The point of the log levels isn't to scare or calm but to be filtered easily. If I come back after a month of running the tool, I would like to be able to look at one file and say "what serious problem happened over this month, filter by ERROR or above". So the question is: should this event be in such a file or not. I think it should. Do you think it should not?

Reply:
> what serious problem happened over this month, filter by ERROR or above

Hmm, my intuition would be to filter by WARNING or above, but I'm not too sure about that. My feeling is: I'm not sure if a node going down is necessarily a "serious problem" (requiring the ERROR level) for the f-API to worry about, as the eventual hope is that individual nodes are managed in a de-centralized way, right? So, if for example the Douglas node becomes unreachable, we should see a warning, but the ERROR should be on their end, not ours - b/c there's not much we can do. Unless we plan to do something with the ERROR on our side?

Reply:
To clarify: all of the logs (info / warning / error) could be sent to the same log file right? If not, we always have the option to filter by warnings or above?

Reply:
yes, we/logging can always filter. I just read through again - we do use warnings for node response failures elsewhere - then let's use warnings consistently